kernel/kernel-api/src/main/java/io/delta/kernel/engine/CommitCoordinatorClientHandler.java

```diff
@@ -0,0 +1,152 @@
+     */
+    CommitResponse commit(
+            String logPath,
+            Map<String, String> tableConf,
+            long commitVersion,
+            CloseableIterator<Row> actions,
+            UpdatedActions updatedActions);
+
+    /**
+     * API to get the unbackfilled commits for the table represented by the given logPath.
+     * Commits older than startVersion or newer than endVersion (if given) are ignored. The
+     * returned commits are contiguous and in ascending version order.
+     *
+     * Note that the first version returned by this API may not be equal to startVersion. This
+     * happens when some versions starting from startVersion have already been backfilled and so
+     * the commit coordinator may have stopped tracking them.
+     *
+     * The returned latestTableVersion is the maximum commit version ratified by the commit
+     * coordinator. Note that returning latestTableVersion as -1 is acceptable only if the commit
+     * coordinator never ratified any version, i.e. it never accepted any unbackfilled commit.
+     *
+     * @param tablePath The path to the delta log of the table for which the unbackfilled
+     *                  commits should be retrieved.
+     * @param tableConf The table configuration that was returned by the commit coordinator
+     *                  during registration.
+     * @param startVersion The minimum version of the commit that should be returned. Can be null.
+     * @param endVersion The maximum version of the commit that should be returned. Can be null.
+     * @return GetCommitsResponse which has a list of {@link Commit}s and the latestTableVersion
+     *         which is tracked by {@link CommitCoordinatorClientHandler}.
+     */
+    GetCommitsResponse getCommits(
+            String tablePath,
+            Map<String, String> tableConf,
+            Long startVersion,
+            Long endVersion);
+
+    /**
+     * API to ask the commit coordinator client to backfill all commits up to {@code version}
+     * and notify the commit coordinator.
+     *
+     * If this API returns successfully, that means the backfill must have been completed, although
+     * the commit coordinator may not be aware of it yet.
+     *
+     * @param logPath The path to the delta log of the table that should be backfilled.
+     * @param tableConf The table configuration that was returned by the commit coordinator
+     *                  during registration.
+     * @param version The version till which the commit coordinator client should backfill.
+     * @param lastKnownBackfilledVersion The last known version that was backfilled before this API
+     *                                   was called. If it is None or invalid, then the commit
+     *                                   coordinator client should backfill from the beginning of
+     *                                   the table. Can be null.
+     */
+    void backfillToVersion(
```

Comment:
same comment as the one on `registerTable`

Reply:
I think it's the interface API of `CommitCoordinatorClient` defined in the Storage module?